config/environments/production.rb

```diff
@@ -79,17 +79,17 @@
   # Use default logging formatter so that PID and timestamp are not suppressed.
   config.log_formatter = ::Logger::Formatter.new
   config.action_mailer.smtp_settings = {
-    address: Rails.application.secrets.smtp_address,
-    port: Rails.application.secrets.smtp_port,
-    authentication: Rails.application.secrets.smtp_authentication,
-    user_name: Rails.application.secrets.smtp_username,
-    password: Rails.application.secrets.smtp_password,
-    domain: Rails.application.secrets.smtp_domain,
-    enable_starttls_auto: Rails.application.secrets.smtp_starttls_auto,
+    address: Decidim::Env.new('SMTP_ADDRESS').to_s,
+    port: Decidim::Env.new('SMTP_PORT', 587).to_i,
+    authentication: Decidim::Env.new('SMTP_AUTHENTICATION', 'plain').to_s,
+    user_name: Decidim::Env.new('SMTP_USERNAME').to_s,
+    password: Decidim::Env.new('SMTP_PASSWORD').to_s,
+    domain: Decidim::Env.new('SMTP_DOMAIN').to_s,
+    enable_starttls_auto: Decidim::Env.new('SMTP_STARTTLS_AUTO').to_boolean_string,
```

Comment:
Heads-up, this can break SMTP, check out https://github.com/decidim/decidim/pull/15192